database/test/test_triggers.py

```diff
@@ -0,0 +1,233 @@
+
+
+def test_update_lifecycle_status_triggers(
+    session: Session,
+    plan_instance: models.Plan,
+    plan_regulation_group_instance: models.PlanRegulationGroup,
+    land_use_area_instance: models.LandUseArea,
+    other_area_instance: models.OtherArea,
+    line_instance: models.Line,
+    land_use_point_instance: models.LandUsePoint,
+    other_point_instance: models.OtherPoint,
+    plan_regulation_instance: models.PlanRegulation,
+    plan_proposition_instance: models.PlanProposition,
+    code_instance: codes.LifeCycleStatus,
+    another_code_instance: codes.LifeCycleStatus,
+):
+    # Set common plan_regulation_group
+    plan_instance.plan_regulation_group = plan_regulation_group_instance
+    land_use_area_instance.plan_regulation_group = plan_regulation_group_instance
+    other_area_instance.plan_regulation_group = plan_regulation_group_instance
+    line_instance.plan_regulation_group = plan_regulation_group_instance
+    land_use_point_instance.plan_regulation_group = plan_regulation_group_instance
+    other_point_instance.plan_regulation_group = plan_regulation_group_instance
+    plan_regulation_instance.plan_regulation_group = plan_regulation_group_instance
+    plan_proposition_instance.plan_regulation_group = plan_regulation_group_instance
+    session.flush()
+
+    # Set common lifecycle_status
+    land_use_area_instance.lifecycle_status = code_instance
+    other_area_instance.lifecycle_status = code_instance
+    line_instance.lifecycle_status = code_instance
+    land_use_point_instance.lifecycle_status = code_instance
+    other_point_instance.lifecycle_status = code_instance
+    plan_regulation_instance.lifecycle_status = code_instance
+    plan_proposition_instance.lifecycle_status = code_instance
+    plan_instance.lifecycle_status = code_instance
+    session.flush()
+    assert plan_instance.lifecycle_status is code_instance
+
+    # Change lifecycle status
+    plan_instance.lifecycle_status = another_code_instance
+    session.commit()
+
+    assert plan_instance.lifecycle_status_id == another_code_instance.id
+    assert land_use_area_instance.lifecycle_status_id == another_code_instance.id
+    assert other_area_instance.lifecycle_status_id == another_code_instance.id
+    assert line_instance.lifecycle_status_id == another_code_instance.id
+    assert land_use_point_instance.lifecycle_status_id == another_code_instance.id
+    assert other_point_instance.lifecycle_status_id == another_code_instance.id
+    assert plan_regulation_instance.lifecycle_status_id == another_code_instance.id
+    assert plan_proposition_instance.lifecycle_status_id == another_code_instance.id
+
+
+def test_add_plan_id_fkey_triggers(
+    session: Session,
+    plan_instance: models.Plan,
+    code_instance: codes.LifeCycleStatus,
+    type_of_underground_instance: codes.TypeOfUnderground,
+    plan_regulation_group_instance: models.PlanRegulationGroup,
+):
+    # Create new plan objects without plan_id
```

Comment:
I'm thinking we should actually test with *two* plan instances in different locations, and always check that the object is added to the plan instance that contains that area.

Also, I'm thinking this trigger will actually have to function as a constraint, I guess that can be done with is_constraint=True.

This means that if no plan is found that contains the whole area, PostGIS should return an error before saving the object. So I guess the trigger will have to be run before saving.